File: tests/clapi/functional/resources/code/php/example/src/Shared/AggregateRoot.php
<?php

declare(strict_types=1);

namespace App\Shared;

final class AggregateRoot
{
    private array $domainEvents = [];

    final public function pullDomainEvents(): array
    {
        $domainEvents = $this->domainEvents;
        $this->domainEvents = [];

        return $domainEvents;
    }

    final protected function record(mixed $domainEvent): void
    {
        $this->domainEvents[] = $domainEvent;
    }

}

File: tests/clapi/functional/resources/code/java/example/src/com/example/shared/AggregateRoot.java
package com.example.shared;

import java.util.ArrayList;
import java.util.List;

public class AggregateRoot {
    private List<Object> domainEvents = new ArrayList<>();

    protected final void record(Object domainEvent) {
        this.domainEvents.add(domainEvent);
    }

    public final List<Object> pullDomainEvents() {
        List<Object> events = new ArrayList<>(this.domainEvents);
        this.domainEvents.clear();
        return events;
    }
}
